Apps/W1/EDocumentConnectors/SignUp/app/src/APIRequests.Codeunit.al

```diff
@@ -0,0 +1,221 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.EServices.EDocumentConnector.SignUp;
+
+using Microsoft.EServices.EDocument;
+using Microsoft.Foundation.Company;
+using Microsoft.Sales.Customer;
+using System.Security.Authentication;
+using System.Text;
+using System.Utilities;
+using System.Xml;
+
+codeunit 6380 APIRequests
+{
+    Access = Internal;
+
+    // https://<BASE URL>/api/Peppol
+    procedure SendFilePostRequest(var TempBlob: Codeunit "Temp Blob"; EDocument: Record "E-Document"; var HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage): Boolean
+    var
+        ConnectionSetup: Record ConnectionSetup;
+        Payload: Text;
+        ContentHttpHeaders: HttpHeaders;
+        HttpContent: HttpContent;
+        ContentText: Text;
+        UriTemplateLbl: Label '%1/api/Peppol', Comment = '%1 = Service Url', Locked = true;
```

Comment:
Does not follow style consitency for repo. Please move to global var

Reply:
I know this was argued in a previous comment. And I argue against it again as every UrlTemplate is different per call. Also, as the Url is different per procedure call, having it as a global makes the procedure less self-contained, I mean the UrlTemplate is only used here and is intimiately connected to the string substitution in the procedure.

So, the only thing I can do is to move all UrlTemplate to be global and then rename them related to the procedure they are used in. I still feel it gets less clear about the composition of the final url as the UrlTemplate then is not local.

I'll see what I can do to keep most of it global but keep the details local. Let me think about this...

Reply:
Hmm, i see you point. 
Looking at Avalara to remind myself how it was done there it ended being:
`this.HttpRequestMessage.SetRequestUri(this.BaseUrl + '/einvoicing/documents/' + Id + '/$download');`

Which i think is more clear than using a label anyway. We dont care about translations

Reply:
As long as it is pretty easy to understand the constructed url.

Reply:
Just looked around in the Outlook REST API that we looked at historically to use for integration. You are correct, the UrlTemplates are all global. I will adjust accordingly.